### ISSUE TITLE
Project structure clarification

### ISSUE THREAD

**Issue by DRSDavidSoft** [NativePHP/electron-plugin#7]
### Question
So basically, this project has two components according to my understanding, and PHP is used in combination with Electron.js (or later Tauri), and the actual OS APIs are consumed by Electron in this alpha stage. PHP doesn't use FFI or some other technologies to communicate with the shell.

This gives me two questions:
1. First, surely you can use this electron-plugin repo with other backend technologies than PHP, as long as they are able to make POST requests here?
2. Wouldn't you prefer to implement FFI and/or COM objects in PHP to communicate with the OS, such as Windows? For example, PHP supports communication with WMI and the shell objects, which seems mightly more efficient than inter-process communication by the means of HTTP APIs.

**Comment by mpociot** [comment]
> First, surely you can use this electron-plugin repo with other backend technologies than PHP, as long as they are able to make POST requests here?

That's correct. As of right now, the electron-plugin package simply provides a HTTP API that any technology can consume.

> Wouldn't you prefer to implement FFI and/or COM objects in PHP to communicate with the OS, such as Windows? For example, PHP supports communication with WMI and the shell objects, which seems mightly more efficient than inter-process communication by the means of HTTP APIs.

I don't think that the overhead of a local HTTP API call is very noticeable. 
Do you mean that the PHP binary communicates directly with the electron binary in order to avoid the HTTP roundtrip?
